bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -234,17 +234,19 @@ public void run() {
         } else {
             this.systemBroadcastReceiver = null;
         }
-        connectivity.registerForNetworkChanges();
 
         registerOrientationChangeListener();
 
+        // initialise plugins before attempting to flush any errors
+        loadPlugins(configuration);
+
         // Flush any on-disk errors
         eventStore.flushOnLaunch();
+        connectivity.registerForNetworkChanges();
```

Comment:
We seem to be left with a seemingly illogical order - do the positions of the two event registering lines (connectivity and orientation) impact the issue you're seeing? Logically, I would expect them both to be done together.

Reply:
The orientation listener doesn't affect whether errors are flushed and has therefore been left as-is to minimise change. L244/245 are the only places in the `Client` constructor that trigger delivery of error reports so have therefore been moved after the plugin loading to avoid any potential race.

This is unrelated to the React Native flake issue but is worth addressing nonetheless.

Reply:
So why move `registerForNetworkChanges`? I can see this affects the flushing of reports as a connection may not be present. Why is it desirable not to have this updated until after the flush attempt?

Reply:
I'll switch L245 with L244 - the important thing is that both occur after plugins are loaded

Reply:
But I'm still not seeing any justification for moving `registerForNetworkChanges`. I'd like for it either to be moved back to its original location so there's no net change, or if it's key to the change, it should be mentioned in the PR or comments why it should be after  loadPlugins for future reference.

Reply:
`registerForNetworkChanges` can invoke the following callback which triggers error flushing, so it's therefore essential to call `loadPlugins` before this: https://github.com/bugsnag/bugsnag-android/blob/82b8236c42099b68920fff1f353cab34d85f1f86/bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java#L127

Reply:
OK - ☝️ , that's the detail I was looking for.